newrelic/core/config.py

```diff
@@ -1103,53 +1102,71 @@ def ignore_status_code(status):
 def is_expected_error(
     exc_info,
     status_code=None,
+    settings=None,
 ):
+    """Check if an error is expected based on rules matching. Default is False when settings lookup fails."""
     return error_matches_rules(
         "expected",
         exc_info,
         status_code=status_code,
+        settings=settings,
     )
 
 
 def should_ignore_error(
     exc_info,
     status_code=None,
+    settings=None,
 ):
+    """Check if an error should be ignored based on rules matching. Default is True when settings lookup fails."""
     return error_matches_rules(
         "ignore",
         exc_info,
         status_code=status_code,
+        settings=settings,
     )
 
 
 def error_matches_rules(
     rules_prefix,
     exc_info,
     status_code=None,
+    settings=None,
 ):
+    """
+    Attempt to match exception to rules based on prefix.
+
+    rules_prefix is one of [ignore, expected]
+    exc_info is an exception tuple of (exc, val, tb)
+    status_code is an optional value or callable taking in exc_info that returns an int-like object
+    origin is either the current application or trace.
+    """
     # Delay imports to prevent lockups
-    from newrelic.api.application import application_instance
     from newrelic.core.trace_cache import trace_cache
 
-    trace = trace_cache().current_trace()
-    settings = trace and trace.settings
-
-    if not settings:
-        # Retrieve application settings
-        application = application_instance()
-        settings = application and application.settings
-
-    # Default to global settings
-    settings = settings or global_settings()
-
     if not settings:
-        return False
+        # Pull from current transaction if no settings provided
+        tc = trace_cache()
+        transaction = tc.current_transaction()
+        settings = transaction and transaction.settings
+
+        if not settings:
+            # Pull from active trace if no settings on transaction
+            trace = tc.current_trace()
+            settings = trace and trace.settings
+
+            if not settings:
+                # Unable to find rules to match with
+                _logger.error(
+                    "Failed to retrieve exception rules: No settings supplied, or found on transaction or trace."
```

Comment:
Should we fallback to `global_settings()` if none are found on the txn or trace?

Reply:
I wouldn't, I don't want any chances of a bug where serverside overrides are ignored.